x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/QueryFeatureExtractor.java

```diff
@@ -55,11 +55,16 @@ public void setNextReader(LeafReaderContext segmentContext) throws IOException {
             }
             scorers.add(scorer);
         }
-        rankerIterator = new DisjunctionDISIApproximation(disiPriorityQueue);
+
+        rankerIterator = disiPriorityQueue.size() > 0 ? new DisjunctionDISIApproximation(disiPriorityQueue) : null;
```

Comment:
In some condition (all scorers were null), the `rankerIterator` with an empty `disiPriorityQueue` causing the error.